libres/tests/analysis/test_update.cpp

```diff
@@ -0,0 +1,203 @@
+        this->b = enkf_util_rand_normal(b_true - 0.5 * b_std, b_std, rng);
+    }
+
+    double eval(double x) const { return this->a * x + this->b; }
+};
+
+SCENARIO("Running analysis update without row scaling on linear model",
+         "[analysis]") {
+
+    GIVEN("Fixed prior and measurements") {
+
+        analysis_module_type *enkf_module = analysis_module_alloc("STD_ENKF");
+        analysis_module_set_var(enkf_module, ENKF_TRUNCATION_KEY_, "1.0");
+        rng_type *rng = rng_alloc(MZRAN, INIT_DEFAULT);
+
+        int ens_size = 100;
+
+        bool_vector_type *ens_mask = bool_vector_alloc(ens_size, true);
+        meas_data_type *meas_data = meas_data_alloc(ens_mask);
+        obs_data_type *obs_data = obs_data_alloc(1.0);
+
+        model true_model{a_true, b_true};
+
+        std::vector<model> ens;
+        for (int iens = 0; iens < ens_size; iens++)
+            ens.push_back(model(rng));
+
+        // prior
+        int nparam = 2;
+        matrix_type *A = matrix_alloc(nparam, ens_size);
+        for (int iens = 0; iens < ens_size; iens++) {
+            const auto &model = ens[iens];
+            matrix_iset(A, 0, iens, model.a);
+            matrix_iset(A, 1, iens, model.b);
+        }
+        double a_avg_prior = matrix_get_row_sum(A, 0) / ens_size;
+        double b_avg_prior = matrix_get_row_sum(A, 1) / ens_size;
+
+        // observations and measurements
+        int obs_size = 45;
+        std::vector<double> sd_obs_values{10000.0, 100.0, 10.0,   1.0,
+                                          0.1,     0.01,  0.00001};
+        const char *obs_key = "OBS1";
+        meas_block_type *mb =
+            meas_data_add_block(meas_data, obs_key, 1, obs_size);
+        obs_block_type *ob =
+            obs_data_add_block(obs_data, obs_key, obs_size, nullptr, false);
+        std::vector<double> xarg(obs_size);
+        for (int i = 0; i < obs_size; i++) {
+            xarg[i] = i;
+        }
+        for (int iens = 0; iens < ens_size; iens++) {
+            const auto &m = ens[iens];
+            // This is equivalent to M*psi_f in Data Assimilation: The Ensemble Kalman Filter, Geir Evensen, 2009.
+            for (int iobs = 0; iobs < obs_size; iobs++)
+                meas_block_iset(mb, iens, iobs, m.eval(xarg[iobs]));
+        }
+        std::vector<double> measurements(obs_size);
+        for (int iobs = 0; iobs < obs_size; iobs++) {
+            // Add some noise for good measure and a more advanced test
+            // When measurements != true model, then ml estimates != true parameters
+            measurements[iobs] = true_model.eval(xarg[iobs]) +
+                                 enkf_util_rand_normal(0, 1.0, rng);
+        }
+
+        // Leading to fixed Maximum likelihood estimate
+        // It will equal true values when measurements are sampled without noise
+        // It will also stay the same over beliefs
+        double obs_mean = 0.0;
+        double iobs_sum = 0.0;
```

Comment:
In my book variables starting with `i` are indices; maybe `xarg_sum` and `xarg_sum_squared` are better names?

Reply:
We read the same books, agree.